src/bun.js/api/server.zig

```diff
@@ -489,6 +489,9 @@ pub fn NewServer(protocol_enum: enum { http, https }, development_kind: enum { d
         /// So we have to store it.
         user_routes: std.ArrayListUnmanaged(UserRoute) = .{},
 
+        // cache HTMLBundle routes returned by handlers at runtime.
+        html_bundle_route_cache: std.AutoHashMap(*HTMLBundle, bun.ptr.RefPtr(HTMLBundle.Route)) = undefined,
```

Comment:
This should cache the file path used within the HTMLBundle and it should be a bun.StringHashMap so that we deduplicate the HTMLBundle.Route properly